src/mbgl/layout/symbol_instance.hpp

```diff
@@ -28,6 +28,7 @@ class SymbolInstance {
     CollisionFeature textCollisionFeature;
     CollisionFeature iconCollisionFeature;
     WritingModeType writingModes;
+    std::size_t featureIndex;
```

Comment:
^ note: the reason I didn't was because, since `SymbolInstance`:`SymbolFeature` is `many:1`, it seemed like there would have to be some weird acrobatics using `std::shared_ptr`.  Rather than introduce that machinery, I figured since both of these objects are really internal to SymbolLayout, this index-based coupling seemed okay.